src/v/pandaproxy/schema_registry/store.h

```diff
@@ -19,6 +19,23 @@
 
 namespace pandaproxy::schema_registry {
 
+namespace detail {
+
+template<typename T>
+typename T::iterator
+make_non_const_iterator(T& container, typename T::const_iterator it) {
+    return container.erase(it, it);
```

Comment:
this is next level c++

Reply:
> this is next level c++

It's a silly hack. An alternative is to template the container type and have const inferred.